controller/rest/remote_export.go

```diff
@@ -0,0 +1,249 @@
+		log.WithFields(log.Fields{"alias": remoteExportRepository.Alias}).Error(e)
+		restRespErrorMessage(w, http.StatusBadRequest, api.RESTErrInvalidRequest, e)
+		return
+	}
+
+	clusRemoteExportRepository := share.CLUSRemoteExportRepository{
+		RepositoryOwnerUsername:          remoteExportRepository.RepositoryOwnerUsername,
+		RepositoryName:                   remoteExportRepository.RepositoryName,
+		RepositoryBranchName:             remoteExportRepository.RepositoryBranchName,
+		PersonalAccessToken:              remoteExportRepository.PersonalAccessToken,
+		PersonalAccessTokenCommitterName: remoteExportRepository.PersonalAccessTokenCommitterName,
+		PersonalAccessTokenEmail:         remoteExportRepository.PersonalAccessTokenEmail,
+	}
+
+	err = clusHelper.CreateRemoteExportRepository(remoteExportRepository.Alias, &clusRemoteExportRepository, nil)
+	if err != nil {
+		msg := fmt.Sprintf("Could not save remote export repository to kv store: %s", err.Error())
+		restRespErrorMessage(w, http.StatusInternalServerError, api.RESTErrFailWriteCluster, msg)
+		return
+	}
+
+	restRespSuccess(w, r, nil, acc, login, &remoteExportRepository, "Create Remote Export Repository")
+}
+
+func handlerRemoteExportRepositoryGet(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	log.WithFields(log.Fields{"URL": r.URL.String()}).Debug()
+	defer r.Body.Close()
+
+	acc, login := getAccessControl(w, r, "")
+	if acc == nil {
+		return
+	} else if !acc.Authorize(&share.CLUSRemoteExportRepository{}, nil) {
+		restRespAccessDenied(w, login)
+		return
+	}
+
+	alias := ps.ByName("alias")
+	clusRemoteExportRepository, _, err := clusHelper.GetRemoteExportRepository(alias)
+	if err != nil {
+		msg := fmt.Sprintf("Could not retrieve remote export repository from kv store: %s", err.Error())
+		restRespErrorMessage(w, http.StatusInternalServerError, api.RESTErrFailWriteCluster, msg)
+		return
+	}
+
+	remoteExportRepository := api.REST_RemoteExportRepository_GET{
+		RepositoryOwnerUsername:          clusRemoteExportRepository.RepositoryOwnerUsername,
+		RepositoryName:                   clusRemoteExportRepository.RepositoryName,
+		RepositoryBranchName:             clusRemoteExportRepository.RepositoryBranchName,
+		PersonalAccessTokenCommitterName: clusRemoteExportRepository.PersonalAccessTokenCommitterName,
+		PersonalAccessTokenEmail:         clusRemoteExportRepository.PersonalAccessTokenEmail,
+	}
+
+	restRespSuccess(w, r, &remoteExportRepository, acc, login, nil, "Create Remote Export Repository")
+}
+
+func handlerRemoteExportRepositoryDelete(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	log.WithFields(log.Fields{"URL": r.URL.String()}).Debug()
+	defer r.Body.Close()
+
+	acc, login := getAccessControl(w, r, "")
+	if acc == nil {
+		return
+	} else if !acc.Authorize(&share.CLUSRemoteExportRepository{}, nil) {
+		restRespAccessDenied(w, login)
+		return
+	}
+
+	alias := ps.ByName("alias")
+	err := clusHelper.DeleteRemoteExportRepository(alias)
```

Comment:
No lock protection for deletion? See handlerSystemWebhookDelete()

Reply:
As discussed.